app/components/post_body_additional_content/post_body_additional_content.js

```diff
@@ -291,6 +290,46 @@ export default class PostBodyAdditionalContent extends PureComponent {
         return previewComponent ? {...previewComponent.props} : {};
     };
 
+    getYouTubeTime = (link) => {
+        const timeRegex = /[\\?&](t|start|time_continue)=([0-9]+h)?([0-9]+m)?([0-9]+s?)/;
+
+        const time = link.match(timeRegex);
+        if (!time || !time[0]) {
+            return 0;
+        }
+
+        const hours = time[2] ? time[2].match(/([0-9]+)h/) : null;
+        const minutes = time[3] ? time[3].match(/([0-9]+)m/) : null;
+        const seconds = time[4] ? time[4].match(/([0-9]+)s?/) : null;
+
+        let ticks = 0;
+
+        if (hours && hours[1]) {
+            ticks += parseInt(hours[1], 10) * 3600;
+        }
+
+        if (minutes && minutes[1]) {
+            ticks += parseInt(minutes[1], 10) * 60;
+        }
+
+        if (seconds && seconds[1]) {
+            ticks += parseInt(seconds[1], 10);
+        }
+
+        return ticks;
+    };
+
+    getYouTubeVideoId = (link) => {
+        const regex = /(youtube\.com\/watch|(v=)|youtu\.be\/)([a-zA-Z0-9_-]+)/;
```

Comment:
overall looks good to me. just one question though.  Youtube URLs are versatile like should we support embed urls as well?

Ex: https://www.youtube.com/embed/xqCoNej8Zxo

Reply:
@sudheerDev I've updated the regex